src/rp2_common/pico_unique_id/unique_id.c

```diff
@@ -26,3 +26,16 @@ static void __attribute__((constructor)) _retrieve_unique_id_on_boot(void) {
 void pico_get_unique_board_id(pico_unique_board_id_t *id_out) {
     *id_out = retrieved_id;
 }
+
+void pico_get_unique_board_id_string(char *id_out, size_t len) {
+    pico_unique_board_id_t temp = retrieved_id;
+    size_t i;
+    // Generate hex one nibble at a time
+    for (i = 0; (i < len - 1) && (i < PICO_UNIQUE_BOARD_ID_SIZE_BYTES * 2); i++) {
+        size_t bi = i / 2;
+        int nibble = (temp.id[bi] >> 4) & 0x0F;
+        temp.id[bi] = (uint8_t)(temp.id[bi] << 4);
```

Comment:
Apologies if this is a stupid question (I'm not a low-level C programmer), but is this modifying the bytes in the underlying `retrieved_id` ?

Reply:
No, this works on the bytes in`temp` which are copied in line 31.

Reply:
this does feel a bit weird though, i'd be tempted to just do
```
nibble = (temp.id[i/2] >> (4 * (i&1))) & 0xf;
```

Reply:
(I guess @kilograham 's suggestion doesn't modify `temp`, in which case perhaps there's no need for the copy and you can just access `retrieved_id`?)

Reply:
That's a much better way and it doesn't need the copy, thanks.